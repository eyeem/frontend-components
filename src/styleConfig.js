/* @flow */
export const textSizes = {
  default: {
    font: {
      small: 14,
      medium: 16,
      large: 16,
      xlarge: 16
    },
    letterSpacing: 0.4
  },
  pStatic: {
    font: {
      small: 14,
      medium: 14,
      large: 14,
      xlarge: 14
    },
    letterSpacing: 0
  },
  pMin: {
    font: {
      small: 12,
      medium: 12,
      large: 12,
      xlarge: 12
    },
    letterSpacing: 0
  },
  headline: {
    font: {
      small: 24,
      medium: 32,
      large: 40,
      xlarge: 56
    },
    letterSpacing: 0.3
  },
  headline2: {
    font: {
      small: 20,
      medium: 24,
      large: 32,
      xlarge: 40
    },
    letterSpacing: 0.3
  },
  headline2Pumped: {
    font: {
      small: 24,
      medium: 28,
      large: 36,
      xlarge: 44
    },
    letterSpacing: 0.3
  },
  headline3: {
    font: {
      small: 18,
      medium: 22,
      large: 28,
      xlarge: 32
    },
    letterSpacing: 0.3
  },
  headline4: {
    font: {
      small: 20,
      medium: 20,
      large: 24,
      xlarge: 24
    },
    letterSpacing: 0.3
  },
  sub1: {
    font: {
      small: 16,
      medium: 20,
      large: 24,
      xlarge: 24
    },
    letterSpacing: 0.3
  },
  sub2: {
    font: {
      small: 14,
      medium: 24,
      large: 24,
      xlarge: 24
    },
    letterSpacing: 0
  },
  subHeader: {
    font: {
      small: 12,
      medium: 12,
      large: 14,
      xlarge: 14
    },
    letterSpacing: 0
  },
  loudSub: {
    font: {
      small: 14,
      medium: 14,
      large: 14,
      xlarge: 16
    },
    letterSpacing: 0.3
  },
  loud: {
    font: {
      small: 56,
      medium: 48,
      large: 56,
      xlarge: 64
    },
    letterSpacing: 0.3
  },
  cap: {
    font: {
      small: 12,
      medium: 14,
      large: 14,
      xlarge: 14
    },
    letterSpacing: 0
  },
  tableP: {
    font: {
      small: 14,
      medium: 14,
      large: 14,
      xlarge: 16
    },
    letterSpacing: 0.3
  },
  sidebarP: {
    font: {
      small: 12,
      medium: 12,
      large: 12,
      xlarge: 14
    },
    letterSpacing: 0
  },
  footerSection: {
    font: {
      small: 16,
      medium: 16,
      large: 16,
      xlarge: 16
    },
    letterSpacing: 0.2
  },
  footerSub: {
    font: {
      small: 10,
      medium: 10,
      large: 10,
      xlarge: 10
    },
    letterSpacing: 0.1
  },
  tablePrice: {
    font: {
      small: 40,
      medium: 20,
      large: 20,
      xlarge: 20
    },
    letterSpacing: 0.2
  }
};

export const textFonts = ['regular', 'light', 'bold'];
export const textDisplays = ['block', 'inline-block', 'inline'];
export const textAligns = ['left', 'center', 'right'];
export const textTopMargins = {
  none: {
    small: 0,
    medium: 0,
    large: 0,
    xlarge: 0
  },
  regular: {
    small: 36,
    medium: 56,
    large: 56,
    xlarge: 56
  },
  xsmall: {
    small: 4,
    medium: 4,
    large: 4,
    xlarge: 8
  },
  xsmall2: {
    small: 0,
    medium: 0,
    large: 2,
    xlarge: 4
  },
  small: {
    small: 16,
    medium: 24,
    large: 24,
    xlarge: 24
  },
  smallStatic: {
    small: 16,
    medium: 16,
    large: 16,
    xlarge: 16
  },
  small2: {
    small: 12,
    medium: 12,
    large: 12,
    xlarge: 12
  },
  medium: {
    small: 45,
    medium: 68,
    large: 76,
    xlarge: 76
  },
  large: {
    small: 48,
    medium: 48,
    large: 48,
    xlarge: 48
  },
  large2: {
    small: 40,
    medium: 40,
    large: 40,
    xlarge: 40
  },
  xlarge: {
    small: 104,
    medium: 104,
    large: 104,
    xlarge: 104
  }
};
export const textColors = {
  default: { regular: 'blacks.black5' },
  green: { regular: 'solidColors.green' },
  white: { regular: 'whites.white' },
  grey1: { regular: 'greys.grey1' },
  grey2: { regular: 'greys.grey2' },
  grey3: { regular: 'greys.grey3' },
  opacityWhite: {
    regular: 'uncategorized.unnamed28'
  },
  link: {
    regular: 'greys.grey3',
    hover: 'greys.grey4',
    active: 'blacks.black5'
  },
  linkOnDark: {
    regular: 'greys.grey4',
    hover: 'whites.white',
    active: 'greys.grey5'
  },
  linkBlack: {
    regular: 'blacks.black5',
    hover: 'uncategorized.unnamed7',
    active: 'blacks.black5'
  },
  linkGreen: {
    regular: 'solidColors.green',
    hover: 'solidColors.greenHover',
    active: 'solidColors.greenActive'
  },
  linkOrange: {
    regular: 'solidColors.orange',
    hover: 'solidColors.orangeHover',
    active: 'solidColors.orangeActive'
  },
  linkRed: {
    regular: 'solidColors.red',
    hover: 'solidColors.redHover',
    active: 'solidColors.redActive'
  },
  linkWhite: {
    regular: 'whites.white',
    hover: 'whites.pearlWhite',
    active: 'whites.pearlWhite'
  }
};

export const buttonColors = {
  default: {
    fontRegular: 'whites.white',
    bgRegular: 'uncategorized.unnamed1',
    bgHover: 'greys.grey4',
    bgActive: 'greys.grey3',
    bgDisabled: 'greys.grey1',
    fontDisabled: 'whites.white'
  },
  green: {
    fontRegular: 'whites.white',
    bgRegular: 'solidColors.green',
    bgHover: 'solidColors.greenHover',
    bgActive: 'solidColors.greenActive',
    bgDisabled: 'solidColors.greenHover',
    fontDisabled: 'whites.white'
  },
  greenFont: {
    fontRegular: 'solidColors.green',
    bgRegular: 'whites.transparent',
    fontHover: 'whites.white',
    bgHover: 'solidColors.green',
    bgActive: 'solidColors.greenActive',
    bgDisabled: 'whites.transparent',
    fontDisabled: 'solidColors.green'
  },
  orange: {
    fontRegular: 'whites.white',
    bgRegular: 'solidColors.orange',
    bgHover: 'solidColors.orangeHover',
    bgActive: 'solidColors.orangeActive',
    bgDisabled: 'solidColors.orangeHover',
    fontDisabled: 'whites.white'
  },
  red: {
    fontRegular: 'whites.white',
    bgRegular: 'solidColors.red',
    bgHover: 'solidColors.redHover',
    bgActive: 'solidColors.redActive',
    bgDisabled: 'greys.grey1',
    fontDisabled: 'whites.white'
  },
  white: {
    fontRegular: 'blacks.black1',
    bgRegular: 'whites.white',
    bgHover: 'uncategorized.unnamed28',
    bgActive: 'uncategorized.unnamed12',
    bgDisabled: 'uncategorized.unnamed28',
    fontDisabled: 'uncategorized.unnamed17'
  },
  black: {
    fontRegular: 'whites.white',
    bgRegular: 'blacks.black1',
    bgHover: 'blacks.black5',
    bgActive: 'blacks.black1',
    bgDisabled: 'blacks.black1',
    fontDisabled: 'whites.white'
  },
  bluePaypal: {
    fontRegular: 'whites.white',
    bgRegular: 'foreigns.paypal',
    bgHover: 'blacks.black1',
    bgActive: 'blacks.black1',
    bgDisabled: 'foreigns.paypal',
    fontDisabled: 'whites.white'
  },
  whiteTrans: {
    fontRegular: 'greys.grey1',
    bgRegular: 'uncategorized.unnamed3',
    bgHover: 'uncategorized.unnamed12',
    bgActive: 'uncategorized.unnamed12',
    bgDisabled: 'uncategorized.unnamed12',
    fontDisabled: 'whites.white'
  },
  blackTrans: {
    fontRegular: 'whites.white',
    bgRegular: 'uncategorized.unnamed10',
    bgHover: 'uncategorized.unnamed8',
    bgActive: 'uncategorized.unnamed8',
    bgDisabled: 'uncategorized.unnamed8',
    fontDisabled: 'whites.white'
  },
  facebook: {
    fontRegular: 'whites.white',
    bgRegular: 'foreigns.facebook2',
    bgHover: 'foreigns.facebook2Hover',
    bgActive: 'foreigns.facebook2Active',
    bgDisabled: 'foreigns.facebook2Disabled',
    fontDisabled: 'whites.white'
  },
  ghostWhite: {
    fontRegular: 'whites.white',
    bgRegular: 'whites.transparent',
    bgDisabled: 'whites.transparent',
    fontDisabled: 'whites.white',
    borderRegular: 'whites.white',
    borderHover: 'whites.white',
    borderActive: 'whites.white'
  },
  ghostGreen: {
    fontRegular: 'solidColors.green',
    bgRegular: 'whites.transparent',
    bgDisabled: 'whites.transparent',
    borderRegular: 'solidColors.green',
    borderHover: 'solidColors.greenHover',
    borderActive: 'solidColors.greenActive',
    fontDisabled: 'solidColors.greenHover',
    borderDisabled: 'solidColors.greenHover'
  },
  ghostGrey: {
    fontRegular: 'blacks.black5',
    bgRegular: 'whites.transparent',
    bgDisabled: 'whites.transparent',
    borderRegular: 'greys.grey4',
    fontDisabled: 'blacks.black5',
    fontHover: 'solidColors.green',
    fontActive: 'solidColors.green',
    borderDisabled: 'blacks.black1',
    borderHover: 'solidColors.green',
    borderActive: 'solidColors.green'
  },
  transparent: {
    fontRegular: 'greys.grey3',
    bgRegular: 'whites.transparent',
    bgDisabled: 'whites.transparent',
    bgHover: 'uncategorized.unnamed15',
    fontHover: 'whites.white',
    bgActive: 'whites.transparent'
  }
};

// prefills some default values if they were omitted in the definitions
export const getButtonColorSet = (color: string) => ({
  borderRegular: 'whites.transparent',
  borderHover: 'whites.transparent',
  borderActive: 'whites.transparent',
  borderDisabled: buttonColors[color].borderRegular || 'whites.transparent',
  fontHover: buttonColors[color].fontRegular,
  fontActive: buttonColors[color].fontRegular,
  fontDisabled: buttonColors[color].fontRegular,
  ...buttonColors[color]
});

export const buttonSizes = {
  default: {
    fontSize: '16px',
    lineHeight: '24px',
    padding: '7px 15px',
    maxHeight: '40px',
    iconSize: '16'
  },
  large: {
    fontSize: '16px',
    lineHeight: '24px',
    padding: '11px 16px 12px 16px',
    maxHeight: '48px',
    iconSize: '18'
  },
  small: {
    fontSize: '14px',
    lineHeight: '16px',
    padding: '7px 15px',
    maxHeight: '40px',
    iconSize: '14'
  },
  minimal: {
    padding: '8px',
    fontSize: '12px',
    lineHeight: '16px',
    maxHeight: '32px',
    iconSize: '16'
  }
};

export const icons = {
  eyeconfont: [
    'star',
    'image',
    'mail',
    'phone',
    'close',
    'timing',
    'prize',
    'remove',
    'alert',
    'flag',
    'embed',
    'delete',
    'backToTop',
    'contactEyeEm',
    'contactFacebook',
    'contactTwitter',
    'contactTumblr'
  ],
  buttons: [
    'cart',
    'checkmark',
    'download',
    'upload',
    'compDownload',
    'like',
    'comment',
    'addToLightbox',
    'isInLightbox',
    'dragndrop',
    'share',
    'facebook'
  ],
  services: ['dropbox', 'googledrive', 'flickr', 'picasa', 'instagram'],
  uncategorized: [
    'leftArrow',
    'rightArrow',
    'leftArrow2',
    'plus',
    'terms',
    'market',
    'growth',
    'star2',
    'star3'
  ]
};
