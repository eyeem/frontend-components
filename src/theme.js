import { BREAKPOINTS } from './styleUtils';

export const colors = {
  whites: {
    white: '#ffffff',
    pearlWhite: '#fafafa',
    transparent: 'rgba(0,0,0,0)'
  },

  greys: {
    grey1: '#666666',
    grey2: '#888888',
    grey3: '#999999',
    grey4: '#cccccc',
    grey5: '#e5e5e5',
    grey6: '#f2f2f2',
    grey7: '#f5f5f5'
  },

  blacks: {
    black1: '#000000',
    black2: '#111111',
    black3: '#1d1d1d',
    black4: '#292929',
    black5: '#333333'
  },

  lightColors: {
    lightRed: '#fff5f5',
    lightBlue: '#e7f0fb'
  },

  solidColors: {
    green: '#22cc88',
    greenv2: '#48B307',
    greenHover: '#4ed69f',
    greenActive: '#1ba36c',
    orange: '#ffaa33',
    orangeHover: '#ffbb5b',
    orangeActive: '#cc8828',
    red: '#ff3333',
    redHover: '#ff5b5b',
    redActive: '#cc2828',
    redv2: '#EA3223'
  },

  foreigns: {
    tumblr: '#34465d',
    twitter: '#55acee',
    facebook1: '#3a5795',
    facebook2: '#3b5998',
    facebook2Hover: '#4c70ba',
    facebook2Disabled: '#96abd6',
    paypal: '#009cde',
    paypalActive: '#007CB1',
    paypalDisabled: '#7FCEEF',
    paypalHover: '#008CC8'
  },

  uncategorized: {
    unnamed1: '#b2b2b2',
    unnamed3: 'rgba(255,255,255,.2)',
    unnamed7: 'rgba(51,51,51,.6)',
    unnamed8: 'rgba(0,0,0,.8)',
    unnamed9: 'rgba(0,0,0,.6)',
    unnamed10: 'rgba(0,0,0,.4)',
    unnamed12: 'rgba(255,255,255,.3)',
    unnamed13: 'rgba(255,255,255,.1)',
    unnamed15: 'rgba(0,0,0,.1)',
    unnamed16: 'rgba(0,0,0,.2)',
    unnamed17: 'rgba(0,0,0,.7)',
    unnamed24: '#ff6666',
    unnamed28: 'rgba(255,255,255,.6)',
    unnamed30: 'rgba(255,255,255,.5)',
    unnamed31: '#5555ff',
    unnamed39: 'rgba(0,0,0,.5)'
  }
};

export const borderRadius = '2px';

/* NOTE: styled-system
         https://styled-system.com/#space-theming
         https://styled-system.com/#responsive-styles
*/
export const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];
export const breakpoints = {
  sm: `${BREAKPOINTS.small}px`,
  md: `${BREAKPOINTS.medium}px`,
  lg: `${BREAKPOINTS.large}px`,
  xl: `${BREAKPOINTS.xlarge}px`
};
