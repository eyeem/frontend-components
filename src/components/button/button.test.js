import React from 'react';
import _ from 'lodash';

import { render, getByText, cleanup, fireEvent } from 'react-testing-library';

import { colors } from '../../theme';
import { buttonColors } from '../../styleConfig';
import Button from './button';

const TESTING_COLOR = 'orange';

afterEach(cleanup);

test('it renders default button', () => {
  const { container } = render(<Button>Hello World</Button>);
  const hasRenderedChildText = getByText(container, 'Hello World');
  const component = container.firstChild;
  expect(hasRenderedChildText).toBeTruthy();
  expect(component).toMatchSnapshot();
});

test('it accepts property disabled', () => {
  const { container } = render(<Button disabled>Disabled Button</Button>);
  const button = container.firstChild;
  expect(button).toHaveProperty('disabled');
  expect(button).toMatchSnapshot();
});

test('it accepts property fullWidth', () => {
  const { container } = render(<Button fullWidth>FullWidth Button</Button>);
  const button = container.firstChild;
  expect(button).toHaveStyleRule('width', '100%');
  expect(button).toMatchSnapshot();
});

test('it accepts property icon=checkmark', () => {
  const { container } = render(<Button icon="checkmark">Icon Button</Button>);
  const button = container.firstChild;
  expect(button.firstChild.nodeName).toBe('svg');
  expect(button).toMatchSnapshot();
});

test('it accepts property color with predefined eyeem color', () => {
  const { container } = render(
    <Button color={TESTING_COLOR}>Color Button</Button>
  );
  const component = container.firstChild;
  const colorSelector = buttonColors[TESTING_COLOR].bgRegular;
  const backgroundColor = _.get(colors, colorSelector);
  expect(component).toHaveStyleRule('background-color', backgroundColor);
  expect(component).toMatchSnapshot();
});

test('it changes color on hover', () => {
  const { container } = render(
    <Button color={TESTING_COLOR}>Hover Button</Button>
  );
  const component = container.firstChild;
  const colorSelector = buttonColors[TESTING_COLOR].bgHover;
  const hoverColor = _.get(colors, colorSelector);
  expect(component).toHaveStyleRule('background-color', hoverColor, {
    modifier: ':hover'
  });
  expect(component).toMatchSnapshot();
});

test('onClick handler beeing called', () => {
  const onClickHandler = jest.fn();
  const { container } = render(
    <Button onClick={onClickHandler}>Click Button</Button>
  );
  const button = container.firstChild;
  fireEvent.click(button);
  expect(onClickHandler).toHaveBeenCalledTimes(1);
});
