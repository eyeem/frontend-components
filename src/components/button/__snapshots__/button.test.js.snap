// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it accepts property color with predefined eyeem color 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 16px;
  line-height: 24px;
  padding: 7px 15px;
  max-height: 40px;
  border: 1px solid rgba(0,0,0,0);
  border-radius: 2px;
  background-color: #ffaa33;
  color: #ffffff;
  fill: #ffffff;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus,
.c0:hover {
  color: #ffffff;
  fill: #ffffff;
  background-color: #ffbb5b;
  border-color: rgba(0,0,0,0);
}

.c0:active {
  background-color: #cc8828;
  border-color: rgba(0,0,0,0);
}

.c0:focus {
  outline: none;
}

<button
  class="c0"
>
  Color Button
</button>
`;

exports[`it accepts property disabled 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 16px;
  line-height: 24px;
  padding: 7px 15px;
  max-height: 40px;
  border: 1px solid rgba(0,0,0,0);
  border-radius: 2px;
  background-color: #b2b2b2;
  color: #ffffff;
  fill: #ffffff;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
  cursor: default;
  border-color: rgba(0,0,0,0);
  background-color: #666666;
  color: #ffffff;
  fill: #ffffff;
}

.c0:focus,
.c0:hover {
  color: #ffffff;
  fill: #ffffff;
  background-color: #cccccc;
  border-color: rgba(0,0,0,0);
}

.c0:active {
  background-color: #999999;
  border-color: rgba(0,0,0,0);
}

.c0:focus {
  outline: none;
}

.c0:focus,
.c0:hover {
  color: #ffffff;
  fill: #ffffff;
  background-color: #666666;
  border-color: rgba(0,0,0,0);
}

.c0:active {
  background-color: #666666;
  border-color: rgba(0,0,0,0);
}

<button
  class="c0"
  disabled=""
>
  Disabled Button
</button>
`;

exports[`it accepts property fullWidth 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 100%;
  font-size: 16px;
  line-height: 24px;
  padding: 7px 15px;
  max-height: 40px;
  border: 1px solid rgba(0,0,0,0);
  border-radius: 2px;
  background-color: #b2b2b2;
  color: #ffffff;
  fill: #ffffff;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus,
.c0:hover {
  color: #ffffff;
  fill: #ffffff;
  background-color: #cccccc;
  border-color: rgba(0,0,0,0);
}

.c0:active {
  background-color: #999999;
  border-color: rgba(0,0,0,0);
}

.c0:focus {
  outline: none;
}

<button
  class="c0"
>
  FullWidth Button
</button>
`;

exports[`it accepts property icon=checkmark 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 16px;
  line-height: 24px;
  padding: 7px 15px;
  max-height: 40px;
  border: 1px solid rgba(0,0,0,0);
  border-radius: 2px;
  background-color: #b2b2b2;
  color: #ffffff;
  fill: #ffffff;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus,
.c0:hover {
  color: #ffffff;
  fill: #ffffff;
  background-color: #cccccc;
  border-color: rgba(0,0,0,0);
}

.c0:active {
  background-color: #999999;
  border-color: rgba(0,0,0,0);
}

.c0:focus {
  outline: none;
}

.c1 {
  margin-right: 8px;
  margin-bottom: -2px;
}

<button
  class="c0"
>
  <svg
    class="c1"
    height="16"
    viewBox="0 0 24 24"
    width="16"
  >
    <path
      d="M2.5 12l6.806 7L21.5 5"
      fill="none"
      stroke="#fff"
      stroke-width="2"
    />
  </svg>
  Icon Button
</button>
`;

exports[`it changes color on hover 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 16px;
  line-height: 24px;
  padding: 7px 15px;
  max-height: 40px;
  border: 1px solid rgba(0,0,0,0);
  border-radius: 2px;
  background-color: #ffaa33;
  color: #ffffff;
  fill: #ffffff;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus,
.c0:hover {
  color: #ffffff;
  fill: #ffffff;
  background-color: #ffbb5b;
  border-color: rgba(0,0,0,0);
}

.c0:active {
  background-color: #cc8828;
  border-color: rgba(0,0,0,0);
}

.c0:focus {
  outline: none;
}

<button
  class="c0"
>
  Hover Button
</button>
`;

exports[`it renders default button 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 16px;
  line-height: 24px;
  padding: 7px 15px;
  max-height: 40px;
  border: 1px solid rgba(0,0,0,0);
  border-radius: 2px;
  background-color: #b2b2b2;
  color: #ffffff;
  fill: #ffffff;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus,
.c0:hover {
  color: #ffffff;
  fill: #ffffff;
  background-color: #cccccc;
  border-color: rgba(0,0,0,0);
}

.c0:active {
  background-color: #999999;
  border-color: rgba(0,0,0,0);
}

.c0:focus {
  outline: none;
}

<button
  class="c0"
>
  Hello World
</button>
`;
